### PR TITLE
consistency on typemax and typemin docs

### PR DESCRIPTION
**Pull request by udohjeremiah**
The `typemax` and `typemin` docs isn't consistent with each other on its examples and `see also` section (see reasons below).

While this isn't an issue when looking up the function with `help>` to work with integers, for floating-point types, its a big deal because:
- On the `typemax` examples, there is a very good example there with a great comment, pointing users to look into `floatmax`, if `typemax` doesn't serves them; but the `typemin` doesn't have this - which isn't great, since both `typemax` and `typemin` have great use.

- The `see also` section on `typemax` point users to other available functions which might prove or introduce other points to consider if `typemax` isn't what they're looking for; for `typemin`, this isn't present.

I provided examples on integers, just so the function isn't thought of like working for only floating points.

Moreover, I feel like the current docs were written assuming users would always lookup `typemin` and `typemax` in `help>`, before using anyone of them. Most time this is true, but I think it's best not to assume.

### CODE REVIEW

File: base/int.jl [hunk 1]
```diff
@@ -772,13 +772,24 @@ promote_rule(::Type{UInt128}, ::Type{Int128}) = UInt128
...
+julia> typemin(UInt32)
+0x00000000
+
 julia> typemin(Float16)
 -Inf16
 
 julia> typemin(Float32)
 -Inf32
+
+julia> nextfloat(-Inf32)  # smallest finite floating point number
```
Comment (giordano) [inline base/int.jl:791]: Why is this here?

Comment (udohjeremiah) [reply base/int.jl:791]: Because `julia> floatmax(Float32)  # largest finite floating point number` is there.

@giordano did you get the point of the PR I was trying to address?
